### ISSUE TITLE
WSL crashes shortly after starting up

### ISSUE THREAD

**Issue by J4yTr1n1ty** [microsoft/WSL#12308]
### Windows Version

Microsoft Windows [Version 10.0.19045.5131]

### WSL Version

2.4.4.0

### Are you using WSL 1 or WSL 2?

- [x] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.167.4-1

### Distro Version

Arch Linux (https://apps.microsoft.com/detail/9mznmnksm73x)

### Repro Steps

# Scenario 1
2. Start WSL from the Terminal
3. Wait for some time

### Expected Behavior

# Scenario 1
The WSL will continue to run until the specified idle time.

# Scenario 2
Kitty will open via RDP

### Actual Behavior

# Scenario 1
WSL crashes shortly after starting up. Only the message [exited] is shown.

# Scenario 2
Kitty does not open

### Diagnostic Logs

Log to start WSL in the terminal with it crashing shortly after: [WslLogs-2024-11-21_12-34-56.zip](https://github.com/user-attachments/files/17845005/WslLogs-2024-11-21_12-34-56.zip)
Log trying to run wslg to open kitty: [WslLogs-2024-11-21_12-31-07.zip](https://github.com/user-attachments/files/17845004/WslLogs-2024-11-21_12-31-07.zip)

**Comment by J4yTr1n1ty** [comment]
After a long time trying out different things, I have "fixed" part of the problem using the following steps:

1. Export the WSL Instance
2. Unregister the WSL Instance
3. Import the WSL Instance
4. Run into WSL as root
5. su to my user

If the default user is set to the personal user, it will still crash

This inhibits the crash. However, GUI applications still do not work.

**Comment by OneBlue** [comment]
@J4yTr1n1ty: What kind of command line do you see "crashing" ? I do see that a process is exiting with error code = 1, but from what I see it's unclear whether this issue is coming from WSL itself, or from that process. The Virtual machine is still running at the end of the first trace you sent.

**Comment by J4yTr1n1ty** [comment]
Hey @OneBlue, what happened for me is that I opened up the WSL, and it showed [exited] after a short while. Since I have not changed anything on my I first assumed it to be a WSL issue. However, it could be very well possible that it's the distribution or a process starting up in the distribution. Is there an easy way to trace this?

**Comment by J4yTr1n1ty** [comment]
An interesting thing I found is that if I remove the default user from the WSL configuration, I can access the virtual machine as root and change my user after the shell is initialized. This completely removes the "crashing" state.

**Comment by Rignchen** [comment]
I have the same problem, we have windows imposed as os on our computer at work so I do almost everything in an Arch WSL2, my WSL often close right after opening, I often have to open it again multiple times until one of them stay open which usually takes me a good 10-30 minutes of work time out of my day >:(
This behavior doesn't seem to be linked to what I'm doing or what my computer is doing

**Comment by Rignchen** [comment]
Hey, I still have this issue, I changed disto since, I now use the NixOs wsl
WSL version 2.4.13.0
I've record this happening
![Image](https://github.com/user-attachments/assets/f99ed507-d5a9-439d-a5ab-1f1fc11ce139)
I do not touch my mouse and keyboard during this recording (except for writing "wsl") but this also happends sometimes when I'm doing stuff can be a problem

**Comment by J4yTr1n1ty** [comment]
I can add to the original that I was using tmux, which was initialized in my zshrc. I'm no longer on WSL so I can't speak to newer updates.

**Comment by OneBlue** [comment]
Interesting. Do you see the same behavior if you run let's say `wsl bash` ?

**Comment by J4yTr1n1ty** [comment]
I tried reproducing my environment as close as possible in a VM, however did not encounter the crash. @Rignchen might be able to try this.

**Comment by anthonymesa** [comment]
I actually have this same issue. I am using Artix Linux with Tmux, and like others, after running wsl from the terminal or cmd, after a short period it crashes as 'exited'. Mostly this seems to only show up for me on first run after wsl loads.

**Comment by 231tr0n** [comment]
This issue is actually reproducible and still happens as of now. I use fish shell with tmux and adding this code to the config.fish file, I can consistently reproduce this issue. This happens because of starting tmux using `exec` command. Also, this happens to me when I freshly start windows and start wsl the first two times and from the third time it does not happen anymore. For anyone facing the same issue, I would suggest you to stop using `exec` to start tmux, and it should be fixed.

```fish
if status is-interactive
    and not set -q TMUX
    # Dont use `exec` to start tmux as it causes wsl to quit
    exec tmux -2u
end
```

**Comment by HamammSoleman** [comment]
yeah the issue definitely is the exec tmux command as @231tr0n pointed out, because after removing that line from my zshrc file, WSL dosen't exit abruptly. What I did notice however, was that if I run my terminal emulator (Alacritty) as an administrator, nothing crashes. Could be something to do with permissions or zsh specifically.

**Comment by lzszone** [comment]
I have the same issue. Ubuntu 24.04, I install it from Store weeks ago.
Just run `wsl` or Ubuntu from start menu or WSL, wait for about 1 minite, it crashes, no tmux for me.
And in my case, reboot Windows can solve it for a short time, once it crashes, it will keep crashing until next reboot.

[WslLogs-2025-12-27_10-16-21.zip](https://github.com/user-attachments/files/24352134/WslLogs-2025-12-27_10-16-21.zip)
